students/ZachCooper/session05/mailroom3.py

```diff
@@ -0,0 +1,233 @@
+        os.mkdir(OUT_PATH)
+
+
+# Donors db which is changed to a dict in main
+def disney_donor_db():
+    return {'yosemite sam': ("Yosemite Sam", [100.56, 200.23, 300]),
+            'yogi bear': ("Yogi Bear", [2000.33, 4000, 1000]),
+            'daffey duck': ("Daffey Duck", [5000.24, 10000.34, 30000]),
+            'wile e. coyote': ("Wile E. Coyote", [300.40, 1000, 750.17]),
+            'bugs bunny': ("Bugs Bunny", [100000.87, 50000, 100]),
+            'mickey mouse': ("Mickey Mouse", [1345.67, 345.78, 1.23]),
+            }
+
+
+# Runs main menu function
+def menu_selection(prompt, dispatch_dict):
+    # Really like the menu dict option using switches
+    while True:
+        response = input(prompt)
+        response = response[:1].lower()
+        if dispatch_dict.get(response, unknown)() == "exit menu":
+            break
+
+
+# First item function in menu
+def thank_you():
+    while True:
+        name = input("Enter a donor's name or 'list' to see all donors or "
+                     "'menu' to exit to main menu\n").strip()
+        if name == "list":
+            print(donor_names())
+        elif name == "menu":
+            return
+        else:
+            take_donation(name)
+
+
+def sort_key(item):
+    return item[1]
+
+
+# Second item function in menu
+def donor_report():
+    """Create donor report"""
+    report_rows = []
+    for (name, gifts) in donors.items():
+        total_gifts = sum(gifts)
+        num_gifts = len(gifts)
+        avg_gift = total_gifts / num_gifts
+        report_rows.append((name, total_gifts, num_gifts, avg_gift))
+
+    report_rows.sort(key=sort_key)
+    print("{:25s} | {:11s} | {:9s} | {:12s}".format(
+        "Donor Name", "Total Given", "Num Gifts", "Average Gift"))
+    print("-" * 66)
+    for row in report_rows:
+        print("{:25s}   ${:11.2f}   {:9d}   ${:12.2f}".format(*row))
+
+
+def find_donor(name):
+    # Strip and lower to counter users entry errors
+    key = name.strip().lower()
+    return donors.get(key)
+
+
+# Third item in menu that returns donor names
+def print_donors_donations():
+    print("Donor name list and donations:\n")
+    # Return donor name along with donation history
+    for donor, value in donors.items():
+        print(value)
+
+
+def donor_names():
+    print("Donor Name:")
+    # Returns only donor names
+    for donor in donors:
+        print(donor)
+
+
+# Use to see if donor exists in dictionary
+def donor_in_dict():
+    don_name = input("Enter a name to see if it exits in donor dictionary:")
+    if don_name.lower() in donors:
+        print("{} is a current donor\n".format(don_name))
+    else:
+        if don_name not in donors:
+            print("{} is not a current donor, please return to menu to start donation"
+                  " process as new donor\n".format(don_name))
+        return
+
+
+# Possible menu selection
+# Currently being used in another function
+def new_donor(name):
+        name = name.strip()
```

Comment:
don't yo uneed tp lower() it as well?

A good reason to have a "normalize" function in a separate place that you can use everywhere.